File: JetpacReloaded/JetpacReloaded/Bonus.cs
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;

namespace JetpacReloaded
{
    class Bonus : BaseObject
    {
        private bool bonusLanded = false;
        private int m_prevFrame = 0;

        public Bonus(Texture2D image)
        {
            m_frame = rand.Next(0, 4);
            m_image = image;
            m_width = image.Width / 5;
            m_height = image.Height + 2;
            m_screenLocation = new Vector2(rand.Next(0, 750), -30);
        }

        public void Update()
        {
            if (!bonusLanded)
            {
                m_screenLocation.Y++;
            }
            m_rect = new Rectangle(m_frame * m_width, 0, m_width, m_height);
        }

        new public void Draw(SpriteBatch spriteBatch)
        {
            Vector2 bonusLocation = new Vector2(m_screenLocation.X, m_screenLocation.Y);
            spriteBatch.Draw(m_image, bonusLocation, m_rect, Color.White, 0, Vector2.Zero, 1.0f, 0, 0);
        }

        public Rectangle BonRect
        {
            get
            {
                return new Rectangle((int)m_screenLocation.X, (int)m_screenLocation.Y, m_width, m_height);
            }
        }

        public bool BonusLanded { set { bonusLanded = value; } }

        public void Reset()
        {
            m_prevFrame = m_frame;
            m_frame = rand.Next(0, 4);
            if (m_frame == m_prevFrame)
            {
                m_frame = (m_frame + 1) % 4;
            }
            m_screenLocation.X = rand.Next(0, 750);
            m_screenLocation.Y = -30;
            bonusLanded = false;
        }
    }
}

File: PyGameJetpac/bonus.py
import pygame
import random
from base_object import BaseObject

class bonus(BaseObject):
    def __init__(self, x, y, position, image, frame, width, height):
        super().__init__(x, y, position, image, frame, width, height)
        self.bonusLanded = False
        self.m_prevFrame = 0
        self.m_width = image.get_width() / 5
        self.m_height = image.get_height() + 2
        self.rect = pygame.Rect(position.x, position.y, image.get_width(), image.get_height())
        self.position = pygame.Vector2(random.randint(0, 750), random.randint(0, 0))

    def update(self) -> None:
        if (not self.bonusLanded):
            self.m_position.y += 1
        self.m_rect = pygame.Rect(self.m_frame * self.m_width, 0, self.m_width, self.m_height)

    def bonus_rectangle(self) -> pygame.Rect:
        return pygame.Rect(self.m_position.x, self.m_position.y, self.m_width, self.m_height)

    def bonus_landed(self, value) -> None:
        self.bonusLanded = value

    def reset(self) -> None:
        self.m_prevFrame = self.m_frame
        self.m_frame = random.randint(0, 4)
        if (self.m_frame == self.m_prevFrame):
            self.m_frame = (self.m_frame + 1) % 4
        self.m_position.x = random.randint(0, 750)
        self.m_position.y = -30
        self.bonusLanded = False
